.github/workflows/pull_request.yml

```diff
@@ -10,15 +10,95 @@ on:
       - 19.x
       - 18.x
       - 17.x
+
 jobs:
   buildAndTest:
-    runs-on: ubuntu-latest
+    name: buildAndTest ${{ matrix.java.c }}/${{ matrix.java.t || matrix.java.c }} (${{ matrix.dist.vm }}) on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ 'ubuntu-latest', 'windows-latest' ]
+        java:
+          - { c: '8', t: '8' } # Latest Java 8
+          - { c: '8.0.282', t: '8.0.282' } # Earliest supported Java 8
+          - { c: '11', t: '11' }
+          - { c: '17', t: '17' }
+          - { c: '18', t: '18' }
+        # - { c: '19-ea', t: '19-ea' } # Will be supported by Gradle 7.5
+        dist:
+          - { name: 'zulu', vm: 'hotspot' }
+          - { name: 'adopt-openj9', vm: 'openj9' }
+        exclude:
+          - # IBM Semeru does not offer Java 18
+            os: 'ubuntu-latest'
+            java: { c: '18', t: '18' }
+            dist: { name: 'adopt-openj9', vm: 'openj9' }
+          - # IBM Semeru does not offer Java 18
+            os: 'windows-latest'
+            java: { c: '18', t: '18' }
+            dist: { name: 'adopt-openj9', vm: 'openj9' }
+          - # Save some CPU (and the environment)
+            os: 'ubuntu-latest'
+            java: { c: '11', t: '11' }
+            dist: { name: 'adopt-openj9', vm: 'openj9' }
+          - # Save some CPU (and the environment)
+            os: 'windows-latest'
+            java: { c: '11', t: '11' }
+            dist: { name: 'adopt-openj9', vm: 'openj9' }
```

Comment:
Should we prioritize testing across various JVM vendors? While it's generally a good practice, it does raise the question of which vendors to prioritize, as it's impractical to test against every single one

Testing with different Operating systems and a single vendor sounds more practical

Reply:
Openj9 is a fundamentally different VM to Hotspot
This is especially important with changes like virtual threads in mind that will have a totally different underlying implementation

Reply:
It might be different, but it should follow the spec at the end of the day, no?

Do you envision(or are aware of) JVM-vendor specific changes for any project? I am curious because doing so would potentially impact portability

Reply:
> It might be different, but it should follow the spec at the end of the day, no?
You cannot find fix implementation discrepancies in the library OR JVM if they are never found.

The OpenTelemtry project has good examples of this, e.g; https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/7304